From 8bf495ef113b272bbcda66cee3217cbd79b4b791 Mon Sep 17 00:00:00 2001
From: Sang Ngo <sang.ngo.yf@renesas.com>
Date: Thu, 6 Nov 2025 17:35:38 +0700
Subject: [PATCH] Update the getting reset control to compatible v6.12.36-cip4 and above

Since commit e4ff24e15ef564d92da6fa0ed940c9755d5e4665, the arguments for
__reset_control_get() has changed. We will adopt this change starting
from kernel v6.12.36-cip4, while maintaining backward compatibility
with older versions. These updates are limited to the Linux CIP branch only.

Signed-off-by: Sang Ngo <sang.ngo.yf@renesas.com>
---
 drivers/gpu/arm/midgard/mali_kbase_core_linux.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/arm/midgard/mali_kbase_core_linux.c b/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
index 3e27069..b4cab9f 100644
--- a/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
+++ b/drivers/gpu/arm/midgard/mali_kbase_core_linux.c
@@ -4687,8 +4687,13 @@ int power_control_init(struct kbase_device *kbdev)
 	dev_dbg(&pdev->dev, "Clocks probed: %u\n", kbdev->nr_clocks);
 
 	for (i = 0; i < BASE_MAX_NR_CLOCKS_REGULATORS; i++) {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 12, 36))
 		kbdev->resets[i] = __of_reset_control_get(kbdev->dev->of_node, NULL, i,
-							false, false, true);
+							RESET_CONTROL_EXCLUSIVE);
+#else
+		kbdev->resets[i] = __of_reset_control_get(kbdev->dev->of_node, NULL, i,
+                                                        false, false, true);
+#endif
 		if (IS_ERR_OR_NULL(kbdev->resets[i])) {
 			err = PTR_ERR(kbdev->resets[i]);
 			kbdev->resets[i] = NULL;
-- 
2.25.1

